lib/services/account.ts

```diff
@@ -0,0 +1,55 @@
+import type { Tables } from '@/types/supabase/database';
+
+import { cookies } from 'next/headers';
+
+import { createSupabaseServerClient } from './supabase/server';
+
+export const fetchAccountAICredits = async (
+  accountId: Tables<'account'>['id'],
+) => {
+  const supabase = createSupabaseServerClient(cookies());
```

Comment:
I thought we could just use `supabase service client` here but:
- supabase is using `fetch`
- using the `service client` makes next.js cache the `fetch` requests in the `fetch cache`
- using `server client` with `cookies()` prevents caching because `cookies` is a dynamic function 👍 

https://github.com/supabase/supabase-js/issues/725

Reply:
I guess it's better to call these functions with the `server client` since they'll be called on behalf of the requesting user. gonna remove `accountId` params as well.

we can use `service client` if we move these stuff to the `edge functions` 👍

Reply:
[91cfff4](https://github.com/experiment-station/beecast/pull/14/commits/91cfff4469c8d4d7a9f52c0d553ea972a1ed59fa)